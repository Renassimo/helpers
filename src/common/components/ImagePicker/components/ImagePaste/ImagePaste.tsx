import Button from '@mui/material/Button';

const ImagePaste = ({ onPaste }: { onPaste: () => void }) => {
  return (
    <Button type="button" fullWidth variant="outlined" onClick={onPaste}>
      ...Or paste from clipboard
    </Button>
  );
};

export default ImagePaste;
