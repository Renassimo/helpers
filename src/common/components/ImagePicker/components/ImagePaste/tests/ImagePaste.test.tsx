import userEvent from '@testing-library/user-event';
import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import ImagePaste from '../ImagePaste';

describe('ImagePaste', () => {
  const mockedOnPaste = jest.fn();

  test('renders snapshot successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <ImagePaste onPaste={mockedOnPaste} />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when clicked', () => {
    test('call onPaste', async () => {
      // Arange
      const { getByText } = renderWithTheme(
        <ImagePaste onPaste={mockedOnPaste} />
      );
      // Act
      await userEvent.click(getByText('...Or paste from clipboard'));
      // Assert
      expect(mockedOnPaste).toHaveBeenCalled();
    });
  });
});
