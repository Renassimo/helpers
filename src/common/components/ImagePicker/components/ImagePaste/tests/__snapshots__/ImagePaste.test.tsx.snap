// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePaste renders snapshot successfully 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-xsqe0f-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      ...Or paste from clipboard
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
